builders/base/buildpack/bin/template.build

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+set -eo pipefail
```

Comment:
Nit: would be nice to add `xu` --> `set -eoux pipefail` to all the bash script. Makes bash debugging easier: https://gist.github.com/mohanpedala/1e2ff5661761d3abd0385e8223e16425

Reply:
so the `u` actually throws errors whenever we add the __Variables file, since not all of the environment variables are necessarily set locally (some are pipeline variables, like BUILD_BUILDNUMBER). Added the others to most scripts, besides the startup script which contains secrets so we don't want to be printing that out